spec/lib/efile/ny/it201_spec.rb

```diff
@@ -39,20 +39,127 @@
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_6].value).to eq(2700)
+        expect(instance.lines[:IT213_LINE_7].value).to eq(877)
+        expect(instance.lines[:IT213_LINE_14].value).to eq(660)
+      end
+    end
+
+    context "when the client has claimed fed_ctc > 0, worksheet A line 8 > worksheet A line 12, and has 3 dependents" do
+      before do
+        intake.dependents.create(dob: 5.years.ago)
+        intake.dependents.create(dob: 3.years.ago)
+        intake.direct_file_data.fed_tax = 0
+        intake.direct_file_data.fed_ctc = 1000
+      end
+
+      it "calculates worksheets and finishes calculations" do
+        instance.calculate
+        expect(instance.lines[:IT213_LINE_1].value).to eq(1)
+        expect(instance.lines[:IT213_LINE_2].value).to eq(1)
+        expect(instance.lines[:IT213_LINE_3].value).to eq(1)
+        expect(instance.lines[:IT213_LINE_4].value).to eq(3) # dependents
+        expect(instance.lines[:IT213_LINE_5].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_1].value).to eq(3000) # 1000 * 1 dependent
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_2].value).to eq(32351)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_3].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_4].value).to eq(32351)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_5].value).to eq(75000)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_6].value).to be_nil
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_7].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_8].value).to eq(3000)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_9].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_10].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_11].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_12].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_A_LINE_13].value).to eq(0)
+        expect(instance.lines[:IT213_LINE_6].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_1].value).to eq(3000)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_2].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_3].value).to eq(3000)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_4A].value).to eq(21000)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_4B].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_5].value).to eq(18000)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_6].value).to eq(2700)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_7].value).to eq(0)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_8].value).to eq(2700)
+        expect(instance.lines[:IT213_WORKSHEET_B_LINE_9].value).to eq(2700)
+        expect(instance.lines[:IT213_LINE_7].value).to eq(2700)
+        expect(instance.lines[:IT213_LINE_14].value).to eq(891)
+      end
+    end
```

Comment:
nonblocking question out of curiosity: should this be in a separate it213 spec file or should this test suite be labeled more generally for ny calculations?

Reply:
yeah good question! tl;dr is i don't feel super strongly about whether it'd be better to rename or do some extra work to initialize the it213/214/215 tests in a different place, but they always require it201 to be calculated first so this was the easiest path to adding test coverage across all the NY forms. 

back when i first tried writing tests for it213/214/215 i started them in their own files but in practice they are always calculated together starting with it201 and many use cases require it201 to be calculated first. i'm sure we could do the work of initializing it201 everywhere but the easiest route was to just test all of the NY calculations together in one place. we could rename it too, although i wasn't sure if breaking the standard naming convention was worth it. since what's going on here is sorta it201 delegating some of its calculations to other forms, i thought it wasn't too much of a stretch to just consider it all it201 testing 🤷